scripts/get_bank_registry_be.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+import json
+import xlrd
+import requests
+
+URL = ("https://www.nbb.be/doc/be/be/protocol/r_fulllist_of_codes_current.xlsx")
+
+
+def process(URL):
+    r = requests.get(URL)
+    book = xlrd.open_workbook(file_contents=r.content)
+    registry = []
+    skip_field = ["VRIJ", "NAV", "NAP", "-", "VRIJ - LIBRE"]
+    row_count = 0
+
+    if len(book.sheets()) == 1:
+        sheet = book.sheets()[0]
+    for i in range(sheet.nrows):
+        if row_count > 1:
+            row_count += 1
+            row = sheet.row_values(i)
+            column = 0
+            registry_entry_template = {"bank_code": None, "bic": None, "country_code": "BE",
+                                       "name": None, "primary": True, "short_name": None}
+            skip_row = False
+            for cell in row:
+                if column == 1:
+                    if cell.upper() not in skip_field:
+                        registry_entry_template["bic"] = str(
+                            cell.upper().replace(" ", ""))
+                    else:
+                        skip_row = True
+                elif column == 0:
+                    registry_entry_template["bank_code"] = str(cell)
+                elif column == 2:
+                    if cell is None:
```

Comment:
Well the reasoning behind the None would be the fact that you sometimes have an empty field in the third column. So in case the third field was empty, it would take the fourth field instead which would contain the name.

Cause if you leave that out, you end up with empty names for some of the bank codes:

`  {
    "country_code": "BE",
    "primary": true,
    "bic": "CREGBEBB",
    "bank_code": "729",
    "name": "",
    "short_name": ""
  }`

Also I am full aware that you can use an if without else, but as I was always tought to be explicit I try to keep it explicitly mentioned. But this shouldn't applicable in this case given the else serves a purpose, no?

Reply:
Well the reasoning behind the None would be the fact that you sometimes have an empty field in the third column. And I did not have a full solution for those couple of exceptions, I thought having something working that works for almost all of them was better than nothing at all. I am under the impression that was a wrong decision.

I am aware that you can use an if without else, but as I was always tought to be explicit I try to keep it explicitly mentioned. Which, granted, I missed earlier in the script with the index sheets.


`  {
    "country_code": "BE",
    "primary": true,
    "bic": "CREGBEBB",
    "bank_code": "729",
    "name": "",
    "short_name": ""
  }`